share/instruments/update_instruments_xml.py

```diff
@@ -319,13 +327,116 @@ def to_list(str):
+with open('instrumentsxml.h', 'w', newline='\n', encoding='utf-8') as f:
+    # Header
+    f.write("/*\n\
+ * SPDX-License-Identifier: GPL-3.0-only\n\
+ * MuseScore-CLA-applies\n\
+ *\n\
+ * MuseScore\n\
+ * Music Composition & Notation\n\
+ *\n\
+ * Copyright (C) 2022 MuseScore BVBA and others\n\
+ *\n\
+ * This program is free software: you can redistribute it and/or modify\n\
+ * it under the terms of the GNU General Public License version 3 as\n\
+ * published by the Free Software Foundation.\n\
+ *\n\
+ * This program is distributed in the hope that it will be useful,\n\
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of\n\
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n\
+ * GNU General Public License for more details.\n\
+ *\n\
+ * You should have received a copy of the GNU General Public License\n\
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.\n\
+ */\n\
+\n")
+
+    # Templates
+    f.write("// Templates\n")
+    d = "../templates"
+    # sort to get same ordering on all platforms
+    for o in sorted(os.listdir(d)):  
+        ofullPath = os.path.join(d, o)
+        if os.path.isdir(ofullPath):
+            templateCategory = o.split("-")[1].replace("_", " ")
+            add_translatable_string(f, 'Templates', templateCategory)
+
+            # sort to get same ordering on all platforms
+            for t in sorted(os.listdir(ofullPath)):
+                if os.path.isdir(os.path.join(ofullPath, t)):
+                    templateName = t.split("-")[1].replace("_", " ")
+                    add_translatable_string(f, 'Templates', templateName)
+
+    f.write("\n")
+    f.write("// Genres\n")
+    for genre in genres.values():
+        add_translatable_string(f, 'InstrumentsXML', genre['name'])
+
+    f.write("\n")
+    f.write("// Families\n")
+    for family in families.values():
+        add_translatable_string(f, 'InstrumentsXML', family['name'])
+
+    f.write("\n")
+    f.write("// Groups & Instruments\n")
+    for group in groups.values():
+        f.write("\n// " + group['name'] + "\n")
+        add_translatable_string(f, 'InstrumentsXML', group['name'])
+
+        for instrument in instruments[group['id']].values():
+            f.write('\n')
+            instrumentId = instrument['id']
+
+            if instrument['traitName'] != '[hide]':
+                for nameType in ['longName', 'shortName', 'trackName']:
+                    add_translatable_string_if_not_null(f, 'InstrumentsXML', instrument[nameType],
+                                                        disambiguation(instrumentId, nameType))
+                
+                add_translatable_string_if_not_null(f, 'InstrumentsXML', instrument['traitName'],
+                                                    disambiguation(instrumentId, 'traitName'), 
+                                                    'traitName codes: * most common for instrument; () hide in score')
```

Comment:
Could you write 'longName', 'shortName' and 'trackName' even if the trait name is '[hide]'? We may need translations of those strings to identify instruments in older scores that don't have instrument IDs. I think we only started saving instrument IDs in MuseScore 3.5 or thereabouts, so in older scores we look at the instrument name. We're not doing this yet but there was a plan to do so, so we'll need them in before the string freeze.

Obviously we don't to translate the trait name if it is '[hide]', so you can keep that in the conditional statement.